[Flaky Test] integration-master: admissionwebhook.TestWebhookTimeoutWithoutWatchCache

<!-- Please only use this template for submitting reports about flaky tests or jobs (pass or fail with no underlying change in code) in Kubernetes CI -->

**Which jobs are flaking**:

- [integration-master](https://testgrid.k8s.io/sig-release-master-blocking#integration-master)

**Which test(s) are flaking**:
- k8s.io/kubernetes/test/integration/apiserver/admissionwebhook.TestWebhookTimeoutWithoutWatchCache
- k8s.io/kubernetes/test/integration/apiserver/admissionwebhook.TestWebhookTimeoutWithoutWatchCache/timed_out_client_requests_skip_later_mutating_webhooks_(regardless_of_failure_policy)_and_fail

**Testgrid link**:
https://testgrid.k8s.io/sig-release-master-blocking#integration-master

**Reason for failure**:
```
W0130 14:27:45.467026  118540 warnings.go:70] flowcontrol.apiserver.k8s.io/v1beta1 FlowSchema is deprecated in v1.23+, unavailable in v1.26+
W0130 14:27:45.470851  118540 warnings.go:70] flowcontrol.apiserver.k8s.io/v1beta1 FlowSchema is deprecated in v1.23+, unavailable in v1.26+
I0130 14:27:46.201015  118540 controller.go:132] OpenAPI AggregationController: action for item : Nothing (removed from the queue).
I0130 14:27:46.201050  118540 controller.go:132] OpenAPI AggregationController: action for item k8s_internal_local_delegation_chain_0000000000: Nothing (removed from the queue).
I0130 14:27:46.209009  118540 storage_scheduling.go:132] created PriorityClass system-node-critical with value 2000001000
I0130 14:27:46.233311  118540 storage_scheduling.go:132] created PriorityClass system-cluster-critical with value 2000000000
I0130 14:27:46.233341  118540 storage_scheduling.go:148] all system priority classes are created successfully or already exist.
W0130 14:27:46.291937  118540 lease.go:233] Resetting endpoints for master service "kubernetes" to [127.0.0.1]
E0130 14:27:46.293134  118540 controller.go:223] unable to sync kubernetes service: Endpoints "kubernetes" is invalid: subsets[0].addresses[0].ip: Invalid value: "127.0.0.1": may not be in the loopback range (127.0.0.0/8)
```
and
```
 === SKIP: test/integration/scheduler_perf TestSchedule100Node3KPods (0.00s)
    scheduler_test.go:74: Skipping because we want to run short tests
=== Failed
=== FAIL: test/integration/apiserver/admissionwebhook TestWebhookTimeoutWithoutWatchCache/timed_out_client_requests_skip_later_mutating_webhooks_(regardless_of_failure_policy)_and_fail (3.03s)
W0130 14:27:50.495109  118540 warnings.go:70] admissionregistration.k8s.io/v1beta1 MutatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 MutatingWebhookConfiguration
W0130 14:27:50.497392  118540 warnings.go:70] admissionregistration.k8s.io/v1beta1 ValidatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 ValidatingWebhookConfiguration
I0130 14:27:53.503357  118540 trace.go:205] Trace[804584285]: "Call mutating webhook" configuration:admission.integration.test-2,webhook:admission.integration.test.0.mutating-1-5s,resource:/v1, Resource=pods,subresource:,operation:CREATE,UID:cbc1fe6d-3939-4587-8349-091a2d830c19 (30-Jan-2021 14:27:50.506) (total time: 2996ms):
Trace[804584285]: [2.99655491s] [2.99655491s] END
W0130 14:27:53.503449  118540 dispatcher.go:170] Failed calling webhook, failing open admission.integration.test.0.mutating-1-5s: failed calling webhook "admission.integration.test.0.mutating-1-5s": Post "https://127.0.0.1:40515/mutating/1/5s?timeout=3s": context deadline exceeded
I0130 14:27:53.512910  118540 tlsconfig.go:255] Shutting down DynamicServingCertificateController
I0130 14:27:53.512934  118540 dynamic_serving_content.go:145] Shutting down serving-cert::/tmp/kubernetes-kube-apiserver186968915/apiserver.crt::/tmp/kubernetes-kube-apiserver186968915/apiserver.key
I0130 14:27:53.512954  118540 dynamic_cafile_content.go:182] Shutting down client-ca-bundle::/tmp/kubernetes-kube-apiserver186968915/client-ca.crt
I0130 14:27:53.513004  118540 secure_serving.go:241] Stopped listening on 127.0.0.1:38969
E0130 14:27:53.516347  118540 controller.go:184] Get "https://127.0.0.1:38969/api/v1/namespaces/default/endpoints/kubernetes": dial tcp 127.0.0.1:38969: connect: connection refused
    --- FAIL: TestWebhookTimeoutWithoutWatchCache/timed_out_client_requests_skip_later_mutating_webhooks_(regardless_of_failure_policy)_and_fail (3.03s)
=== FAIL: test/integration/apiserver/admissionwebhook TestWebhookTimeoutWithoutWatchCache (17.63s) 
```
**Anything else we need to know**:
- Spyglass: https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-integration-master/1355517716575817729


<!-- Please see the deflaking doc (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/flaky-tests.md) for more guidance! -->

--- a/test/integration/apiserver/admissionwebhook/timeout_test.go
+++ b/test/integration/apiserver/admissionwebhook/timeout_test.go
@@ -80,7 +80,7 @@ func testWebhookTimeout(t *testing.T, watchCache bool) {
 		validatingWebhooks []testWebhook
 		expectInvocations  []invocation
 		expectError        bool
-		errorContains      string
+		errorContainsAnyOf []string
 	}{
 		{
 			name:           "minimum of request timeout or webhook timeout propagated",
@@ -133,8 +133,13 @@ func testWebhookTimeout(t *testing.T, watchCache bool) {
 			expectInvocations: []invocation{
 				{path: "/mutating/1/5s", timeoutSeconds: 3}, // from request
 			},
-			expectError:   true,
-			errorContains: "the server was unable to return a response in the time allotted",
+			expectError: true,
+			errorContainsAnyOf: []string{
+				// refer to https://github.com/kubernetes/kubernetes/issues/98606#issuecomment-774832633
+				// for the reason for triggering this scenario
+				"stream error",
+				"the server was unable to return a response in the time allotted",
+			},
 		},
 	}
 
@@ -303,10 +308,18 @@ func testWebhookTimeout(t *testing.T, watchCache bool) {
 				if err == nil {
 					t.Fatalf("expected error but got none")
 				}
-				if tt.errorContains != "" {
-					if !strings.Contains(err.Error(), tt.errorContains) {
-						t.Errorf("expected an error saying %q, but got: %v", tt.errorContains, err)
+
+				expected := false
+				if len(tt.errorContainsAnyOf) != 0 {
+					for _, errStr := range tt.errorContainsAnyOf {
+						if strings.Contains(err.Error(), errStr) {
+							expected = true
+							break
+						}
 					}
+				}
+				if !expected {
+					t.Errorf("expected the error to be any of %q, but got: %v", tt.errorContainsAnyOf, err)
 				}
 				return
 			}

